catalyst/metrics/_segmentation.py

```diff
@@ -151,11 +152,30 @@ def compute_key_value(self) -> Dict[str, torch.Tensor]:
         Returns:
              dict of metrics, including micro, macro and weighted (if weights were given) metrics
         """
+        if self.weights is not None:
+            per_class, micro_metric, macro_metric, weighted_metric = self.compute()
+        else:
+            per_class, micro_metric, macro_metric = self.compute()
+
         metrics = {}
+        for class_idx, value in enumerate(per_class):
+            metrics[
+                f"{self.prefix}{self.metric_name}{self.suffix}/{self.class_names[class_idx]}"
+            ] = value
+
+        metrics[f"{self.prefix}{self.metric_name}{self.suffix}/_micro"] = micro_metric
+        metrics[f"{self.prefix}{self.metric_name}{self.suffix}"] = macro_metric
+        metrics[f"{self.prefix}{self.metric_name}{self.suffix}/_macro"] = macro_metric
+        if self.weights is not None:
+            metrics[f"{self.prefix}{self.metric_name}{self.suffix}/_weighted"] = weighted_metric
+        return metrics
+
+    def compute(self):
```

Comment:
could you please place `compute` on top on `compute_key_value`, as far as `compute_key_value` is using `compute`?